pennylane/transforms/split_tape.py

```diff
@@ -374,32 +266,72 @@ def processing_fn(expanded_results):
 
         return results[0] if len(results) == 1 else results
 
-    return qscripts, processing_fn
+    return tapes, processing_fn
+
+
+def _compute_result_for_each_idx(result, idxs):
+    results = []
+    for idx, coeff in idxs:
+        if coeff is not None:
+            tmp_res = qml.math.convert_like(qml.math.dot(coeff, result), result)
```

Comment:
`convert_like` is probably unnecessary, worst case you can switch coeff and result in the dot.

Reply:
Good catch! Removed `convert_like`

Reply:
Switching coeff and result didn't fix the issue. The `convert_like` was added in [this PR](https://github.com/PennyLaneAI/pennylane/pull/3339).